File: Java/InheritanceExample.java
class A {
    protected int a;

    A() {
    }

    A(int x) {
        a = x;
    }

    void display_a() {
        System.out.println("a = " + a);
    }
}

class B extends A {
    private int b;

    B() {
    }

    B(int y) {
        b = y;
    }

    B(int x, int y) {
        a = x;
        b = y;
    }

    void display_b() {
        System.out.println("b = " + b);
    }

    void display_ab() {
        System.out.println("a = " + a);
        System.out.println("b = " + b);
    }
}

public class InheritanceExample {
    public static void main(String[] args) {
        B obb = new B(10, 20);
        obb.display_ab();
    }
}

File: CPP/InheritanceExample.cpp
#include<iostream>
using namespace std;
class A     //Parent/Base
{
protected:
    int a;
public:
    A() {}
    A(int x)
    {
        a = x;
    }
    void display_a()
    {
        cout<<"a = "<<a<<endl;
    }
};

//Inheritance
class B : public A          //Child/Derived          
{
    int b;
public:
    B() {}
    B(int x)
    {
        b = x;
    }
    B(int x, int y)
    {   
        a = x;
        b = y;
    }
    void display_b()
    {
        cout<<"b = "<<b<<endl;
    }
    void display_ab()
    {
        cout<<"a = "<<a<<endl<<"b = "<<b<<endl;
    }
};


int main()
{
    B obb(10, 20);
    
    obb.display_ab();

    return 0;
}